internal/handlers/transactions_handlers.go

```diff
@@ -122,10 +123,9 @@ func handleTransactionsRequest(c *gin.Context, contractAddress, signature string
 	}
 
 	signatureHash := ""
-	// TODO: implement signature lookup
-	// if signature != "" {
-	// 	signatureHash = crypto.Keccak256Hash([]byte(signature)).Hex()
-	// }
+	if signature != "" {
+		signatureHash = crypto.Keccak256Hash([]byte(signature)).Hex()
+	}
```

Comment:
Does this work with named parameters too? e.g. `Transfer(amount indexed uint256)` ?

Reply:
not like this `Transfer(amount indexed uint256)` but like `Transfer(uint256)`

Reply:
added in the next PR

not sure though if it's the best approach to do this job ourselves instead of giving clients the flexibility to use any signature, along with the responsibility to properly write it